uber/automated_emails.py

```diff
@@ -8,7 +8,11 @@
 class AutomatedEmail:
     instances = OrderedDict()
 
-    extra_models = {}  # extended by plugins
+    queries = {
+        Attendee: lambda session: session.staffers(only_staffing=False),
+        Group: lambda session: session.query(Group).options(subqueryload(Group.attendees))
+    }
+    extra_models = queries  # we've renamed "extra_models" to "queries" but are temporarily keeping the old name for backwards-compatibility
```

Comment:
The emails still aren't fully optimized, but we're doing a better job now than before, both in terms of memory and CPU usage.  The main three changes:

1) We now do intelligent queries with efficient loading strategies.

2) Instead of loading everything into memory, we now iterate over query results one at a time.  This saves us a lot of memory.

3) We do a small sleep after each row so that we don't flood the CPU just with this one thread.  (This adds less than 5 minutes to the wall clock time to do a round of email checks for MAGFest's data set, which seems like a reasonable compromise.)

The memory usage dropped by a few hundred megs and the CPU usage dropped around 50% after these changes.